system/Cache/ResponseCache.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Cache;
+
+use CodeIgniter\HTTP\CLIRequest;
+use CodeIgniter\HTTP\IncomingRequest;
+use CodeIgniter\HTTP\ResponseInterface;
+use Config\Cache as CacheConfig;
+use Exception;
+
+/**
+ * Web Page Caching
+ */
+class ResponseCache
```

Comment:
Any reason not to make this `final`?
```suggestion
final class ResponseCache
```

Reply:
To make it possible to extend it. 
Devs may want to change something.
At least, there is no method to delete cache now.

Reply:
My hunch would be to start it as final, but we can go with this - probably not a lot likely to change in here down the road.

Reply:
I believe it would be better to start as final. If there is a pressing need for this to be overridden then we might consider it as a PR.

Reply:
> If there is a pressing need for this to be overridden then we might consider it as a PR.

Okay. Changed to final.